lib/puppet/rest/client.rb

```diff
@@ -40,10 +42,14 @@ def initialize(route,
     # @param [String] endpoint the endpoint of the configured API to query
     # @param [Hash] query any URL params to add to send to the endpoint
     # @param [Hash] header any additional entries to add to the default header
-    # @return [Puppet::Rest::Response] the response from the server
-    def get(endpoint, query: nil, header: nil)
-      response = @client.get(endpoint, query: query, header: header)
-      Puppet::Rest::Response.new(response)
+    # @yield [HTTPClient::Message, String] the HTTP response and a chunk of its body
```

Comment:
Do we want to expose the upstream client here? Do we use our Response class anymore?

Reply:
We don't have a choice here. The yielding is done by the third-party code. We just pass a reference to the block passed into this method along to it.

And yes, the Response class wrapper is used to wrap `HTTP::Message` objects in thrown exceptions from HTTPClient before rethrowing them.